internal/api/v1/user/contacts/contacts_api.go

```diff
@@ -0,0 +1,140 @@
+package contacts
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/bitcoin-sv/spv-wallet-go-client/commands"
+	"github.com/bitcoin-sv/spv-wallet-go-client/internal/api/v1/user/querybuilders"
+	"github.com/bitcoin-sv/spv-wallet-go-client/queries"
+	"github.com/bitcoin-sv/spv-wallet/models/response"
+	"github.com/go-resty/resty/v2"
+)
+
+const route = "api/v1/contacts"
+
+type API struct {
+	addr       string
+	httpClient *resty.Client
+}
+
+func (a *API) Contacts(ctx context.Context, opts ...queries.ContactQueryOption) (*queries.UserContactsPage, error) {
+	var query queries.ContactQuery
+	for _, o := range opts {
+		o(&query)
+	}
+
+	queryBuilder := querybuilders.NewQueryBuilder([]querybuilders.QueryBuilderOption{
+		querybuilders.WithMetadataFilter(query.Metadata),
+		querybuilders.WithPageFilter(query.PageFilter),
+		querybuilders.WithFilterQueryBuilder(&contactFilterQueryBuilder{
+			contactFilter: query.ContactFilter,
+			modelFilterBuilder: querybuilders.ModelFilterBuilder{
+				ModelFilter: query.ContactFilter.ModelFilter,
+			},
+		}),
+	}...)
```

Comment:
Why dont to put these options right into args?

```suggestion
	queryBuilder := querybuilders.NewQueryBuilder(
		querybuilders.WithMetadataFilter(query.Metadata),
		querybuilders.WithPageFilter(query.PageFilter),
		querybuilders.WithFilterQueryBuilder(&contactFilterQueryBuilder{
			contactFilter: query.ContactFilter,
			modelFilterBuilder: querybuilders.ModelFilterBuilder{
				ModelFilter: query.ContactFilter.ModelFilter,
			},
		}),
	)
```